src/internet_identity/src/main.rs

```diff
@@ -492,6 +501,31 @@ fn authenticate_and_record_activity(anchor_number: AnchorNumber) -> Option<IIDom
     domain
 }
 
+#[derive(Debug)]
+enum IdentityUpdateError {
+    Unauthorized,
+    StorageError(IdentityNumber, StorageError),
+}
+
+impl From<IdentityUpdateError> for String {
+    fn from(err: IdentityUpdateError) -> Self {
+        match err {
+            IdentityUpdateError::Unauthorized => {
+                // This error message is used by the legacy API and should not be changed, even though
+                // it is confusing authentication with authorization.
+                format!("{} could not be authenticated.", caller())
```

Comment:
nit: calling `caller()` in a conversion-to-string function seems weird.  Maybe we should have `IdenityUpdateError::Unauthorized(Principal)` instead? That would also provide more info for debugging/tests.